src/java.base/share/classes/java/util/Objects.java

```diff
@@ -189,6 +190,17 @@ public static String toIdentityString(Object o) {
         return o.getClass().getName() + "@" + Integer.toHexString(System.identityHashCode(o));
     }
 
+   /**
+     * {@return {@code true} if the object is an identity object, otherwise {@code false}}
+     * @param o an object
+     */
+//    @IntrinsicCandidate
+    public static boolean isIdentityObject(Object o) {
+        return o.getClass().isIdentity() ||
+                o.getClass() == Object.class ||
+                !ValhallaFeatures.isEnabled(); // Before Valhalla all objects are identity objects.
```

Comment:
`Class::isIdentity` should check for `!ValhallaFeatures.isEnabled()` as well.   Just a side note: we will need to go through the current implementation to see where this check is needed as well.